test/unittest/extension/axpy_batch_test.cpp

```diff
@@ -0,0 +1,151 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  portBLAS: BLAS implementation using SYCL
+ *
+ *  @filename axpy_batch_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+template <typename scalar_t>
+using combination_t = std::tuple<std::string, index_t, scalar_t, index_t,
+                                 index_t, index_t, index_t, index_t>;
+
+template <typename scalar_t, helper::AllocType mem_alloc>
+void run_test(const combination_t<scalar_t> combi) {
+  std::string alloc;
+  index_t size;
+  scalar_t alpha;
+  index_t incX;
+  index_t incY;
+  index_t stride_mul_x;
+  index_t stride_mul_y;
+  index_t batch_size;
+  std::tie(alloc, size, alpha, incX, incY, stride_mul_x, stride_mul_y,
+           batch_size) = combi;
+
+  const index_t stride_x{size * std::abs(incX) * stride_mul_x};
```

Comment:
This is kind of optional. The stride can start from `(size -1) *(std::abs(incX) + 1) * stride_mul_x`, I think it's worth trying though to evaluate the kernel looping logic over batches specifically when incX>1 and stride_mul==1 (you know  you'll have adjacent values inter-batches & strided values within batches instead of equally distant values everywhere).